R/query_dataset_utils.R

```diff
@@ -0,0 +1,150 @@
+#' Standardise geography inputs
+#'
+#' @description
+#' Create a standard data frame containing geographic_level and location for use by
+#' query_dataset().
+#'
+#' @param geographies String, vector, list or data frame containing the geographic levels and
+#' locations to be queried.
+#'
+#' @return data.frame containing standardised geography specification
+#'
+#' @keywords internal
+#'
+#' @examples
+#' eesyapi:::todf_geographies(c("NAT", "REG"))
+#' eesyapi:::todf_geographies(c("NAT|code|E92000001", "REG|code|E12000001"))
+#' eesyapi:::todf_geographies(c("NAT|code|E92000001", "REG"))
+#' eesyapi:::todf_geographies(
+#'   list(
+#'     geographic_level = c("REG", "LA"),
+#'     location = c("REG|code|E12000001")
+#'   )
+#' )
+#' eesyapi:::todf_geographies(
+#'   data.frame(location = c("REG|code|E12000001", "REG|code|E12000001"))
+#' )
+#' eesyapi:::todf_geographies(
+#'   data.frame(
+#'     geographic_level = c("REG", "LA"),
+#'     location = c("REG|code|E12000001", "REG|code|E12000001")
+#'   )
+#' )
+#' eesyapi:::todf_geographies(
+#'   data.frame(geographic_level = c("NAT", "REG"))
+#' )
+todf_geographies <- function(geographies) {
+  if (is.null(geographies)) {
+    return(NULL)
+  } else if (is.data.frame(geographies)) {
+    if ("location" %in% names(geographies)) {
```

Comment:
Would plural locations make more sense here?

Reply:
That had been niggling at me. Made the shift.